### ISSUE TITLE
[Bug report] In parameter sweep, specifying subsys_update_info does not work as expected when deepcopy = True.

### ISSUE THREAD

**Issue by Harrinive** [scqubits/scqubits#193]
### Describe the bug
When I set deepcopy = True and specify subsys_update_info when initializing the parameter sweep, the sweep can't recognize the subsystem objects in the subsys_update_info. The ParameterSweep object will misinterpret the subsys_update_info and think the bare spectrum don't need to be updated. In the end, the bare spectrum are calculated once and copied for the rest of the sweep, resulting in the incorrect result.

### To Reproduce
```
tmon = scq.Transmon(
    EJ=8.0,
    EC=0.3,
    ng=0.3,
    ncut=50,
    truncated_dim=5,
    id_str="tmon"
)

hspace = scq.HilbertSpace([tmon])

def update_hilbertspace(EJ):
    tmon.EJ = EJ

sweep = scq.ParameterSweep(
    hspace,
    paramvals_by_name={
        "EJ": [7, 8, 9],
    },
    update_hilbertspace=update_hilbertspace,
    evals_count=3,
    subsys_update_info={"EC": ["tmon"]},
    deepcopy=True,
)

print(sweep["evals"])
```
We expect seeing three different list of evals, while the current code gives the repeated eval lists.

**Comment by saipavanc** [comment]
Might be related to #209. It is possible that both are the same issues. This example is perhaps a nice starting point.

**Comment by Harrinive** [comment]
Hey @saipavanc. I made some mistakes in uploading the code to reproduce the error in this issue. The argument in creating the ParameterSweep object should really be `subsys_update_info={"EJ": [tmon]}`. 

Your correction seems to look fine - using `id_str` to identify the subsystem. However, scqubits actually accept the definition of two qubits with the same `id_str` if the user specifies. I feel this is a potential issue and scqubits should not allow it. 

One potential solution is to do a few checks in creating the `ParameterSweep` object, for repetitive `id_str` and illegal `subsys_update_info`.

**Comment by Harrinive** [comment]
Or just for the purpose of a valid parameter sweep, can we just run through the `subsystem_list` of a Hilbert space and find repetitive names?

**Comment by Harrinive** [comment]
it is now fixed in #252

**Comment by saipavanc** [comment]
Thanks @Harrinive !
